src/vs/workbench/api/common/extensionHostMain.ts

```diff
@@ -67,9 +67,10 @@ abstract class ErrorHandler {
 		const map = await extensionService.getExtensionPathIndex();
 		const extensionErrors = new WeakMap<Error, ExtensionIdentifier | undefined>();
 
+		// PART 1
 		// set the prepareStackTrace-handle and use it as a side-effect to associate errors
 		// with extensions - this works by looking up callsites in the extension path index
-		(<any>Error).prepareStackTrace = (error: Error, stackTrace: errors.V8CallSite[]) => {
+		function prepareStackTraceAndFindExtension(error: Error, stackTrace: errors.V8CallSite[]) {
 			let stackTraceMessage = '';
```

Comment:
Small optimization: we should bail if `extensionErrors.has(error)`, since if prepareStackTrace is wrapped more than once (which seems to happen with some library code that Liveshare and Github extensions use) this will be called multiple times.

Reply:
Do you mean code invoking the function directly instead relying on `Error.stack`? In that case, I could save the `findSubstr` calls but still need to return something, right?

Reply:
Oh, true. This is probably fine, very small optimization anyway